packages/ember/tests/routing/query_params_test/shared_state_test.js

```diff
@@ -0,0 +1,88 @@
+import {
+  Controller,
+  Service
+} from 'ember-runtime';
+import Ember from 'ember';
+import { run } from 'ember-metal';
+import { jQuery } from 'ember-views';
+import { QueryParamTestCase, moduleFor } from 'internal-test-helpers';
+
+class SharedStateQPTestCase extends QueryParamTestCase {
+  boot() {
+    this.setupApplication();
+    return this.visitApplication();
+  }
+}
+
+moduleFor('Query Params - shared service state', class extends SharedStateQPTestCase {
+  setupApplication() {
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+      this.route('dashboard');
+    });
+
+    this.application.register('service:filters', Service.extend({
+      shared: true
+    }));
+
+    this.registerController('home', Controller.extend({
+      filters: Ember.inject.service(),
+      queryParams: [
+        { 'filters.shared': 'shared' }
```

Comment:
Based on messing with the Twiddle, this shouldn't be needed. The requirements to reproduce seem to just be that the dependent property of a QP be modified prior to visiting the route of that QP.

So, I think we can simplify this reproduction by not having these query params